test/virtual/host.cpp

```diff
@@ -11,13 +11,143 @@
 
 #include "enclave_impl.h"
 
+/* Temporily set value. */
+#define OE_MAX_ECALLS 256
+
 extern "C"
 {
+    const char* _ecall_table[OE_MAX_ECALLS];
+    uint32_t _ecall_table_size;
     thread_local oe_enclave_t* _enclave;
 
+    // Get the global ecall id from the _ecall_table.
+    static oe_result_t oe_get_global_ecall_id_by_name(
+        const char* name,
+        uint64_t* global_id)
+    {
+        oe_result_t result = OE_NOT_FOUND;
+        uint64_t i;
+
+        if (!global_id)
+        {
+            result = OE_INVALID_PARAMETER;
+            goto done;
+        }
+
+        for (i = 0; i < _ecall_table_size; i++)
+        {
+            if (strcmp(_ecall_table[i], name) == 0)
```

Comment:
If performance is a concern, we can always maintain another sorted array of <name, global-id> pairs and do a binary search, while retaining the same interface.